system_tests/bigquery.py

```diff
@@ -0,0 +1,224 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import operator
+import time
+
+import unittest2
+
+from gcloud import _helpers
+from gcloud.environment_vars import TESTS_PROJECT
+from gcloud import bigquery
+
+
+_helpers.PROJECT = TESTS_PROJECT
+CLIENT = bigquery.Client()
+
+
+class TestPubsub(unittest2.TestCase):
+
+    def setUp(self):
+        self.to_delete = []
+
+    def tearDown(self):
+        for doomed in self.to_delete:
+            doomed.delete()
+
+    def test_create_dataset(self):
+        DATASET_NAME = 'system_tests'
+        dataset = CLIENT.dataset(DATASET_NAME)
+        self.assertFalse(dataset.exists())
+        dataset.create()
+        self.to_delete.append(dataset)
+        self.assertTrue(dataset.exists())
+        self.assertEqual(dataset.name, DATASET_NAME)
+
+    def test_patch_dataset(self):
+        DATASET_NAME = 'system_tests'
+        dataset = CLIENT.dataset(DATASET_NAME)
+        self.assertFalse(dataset.exists())
+        dataset.create()
+        self.to_delete.append(dataset)
+        self.assertTrue(dataset.exists())
+        self.assertEqual(dataset.friendly_name, None)
+        self.assertEqual(dataset.description, None)
+        dataset.patch(friendly_name='Friendly', description='Description')
+        self.assertEqual(dataset.friendly_name, 'Friendly')
+        self.assertEqual(dataset.description, 'Description')
+
+    def test_update_dataset(self):
+        DATASET_NAME = 'system_tests'
+        dataset = CLIENT.dataset(DATASET_NAME)
+        self.assertFalse(dataset.exists())
+        dataset.create()
+        self.to_delete.append(dataset)
+        self.assertTrue(dataset.exists())
+        after = [grant for grant in dataset.access_grants
+                 if grant.entity_id != 'projectWriters']
+        dataset.access_grants = after
+        dataset.update()
+        self.assertEqual(len(dataset.access_grants), len(after))
+        for found, expected in zip(dataset.access_grants, after):
```

Comment:
OK. I suppose we'll find out (via failing tests) if order is not honored by the backend.